packages/nextjs/pages/blockexplorer.tsx

```diff
@@ -0,0 +1,322 @@
+          <ul className="list-disc pl-5 text-neutral-content">
+            {blocks
+              .flatMap(block => block.transactions)
+              .filter(tx => tx.from === searchedAddress || tx.to === searchedAddress)
+              .map((tx, index) => (
+                <li key={index}>{tx.hash}</li>
+              ))}
+          </ul>
+        </div>
+      )}
+      <div className="overflow-x-auto shadow-lg">
+        <table className="min-w-full divide-y divide-primary shadow-lg rounded-lg bg-neutral">
+          <thead className="bg-primary">
+            <tr>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                Transaction Hash
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                Function Called
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                Block Number
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                Time Mined
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                From
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                To
+              </th>
+              <th
+                scope="col"
+                className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+              >
+                Value (ETH)
+              </th>
+            </tr>
+          </thead>
+          <tbody className="bg-base-100 divide-y divide-primary-content text-base-content">
+            {Array.from(blocks.reduce((map, block) => map.set(block.hash, block), new Map()).values()).map(block =>
+              block.transactions.map((tx: TransactionWithFunction) => {
+                const receipt = transactionReceipts[tx.hash];
+
+                const shortTxHash = `${tx.hash.substring(0, 6)}...${tx.hash.substring(tx.hash.length - 4)}`;
+                const date = new Date(block.timestamp * 1000);
+                const month = String(date.getMonth() + 1).padStart(2, "0");
+                const day = String(date.getDate()).padStart(2, "0");
+                const timeMined = `${month}/${day} ${date.toLocaleTimeString("en-US", {
```

Comment:
Since we have global users, best to let their OS/browser settings determine how to display date and timestrings. Forcing `month/day` and US timestring isn't the best approach.

I suggest simplifying this thing into `const timeMined = new Date(block.timestamp * 1000).toLocaleString()`